[2020-12-22 14:11:06,807] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:11:06,824] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:11:06,824] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:11:06,824] {taskinstance.py:867} INFO - Starting attempt 42 of 42
[2020-12-22 14:11:06,824] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:11:06,841] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:11:06,847] {standard_task_runner.py:53} INFO - Started process 58432 to run task
[2020-12-22 14:11:06,909] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:11:06,937] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,937] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:11:06,948] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,947] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:11:06,951] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,950] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:11:06,970] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,969] {batch.py:166} INFO - Batch successfully submitted with id = 38.
[2020-12-22 14:11:06,973] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,973] {batch.py:80} INFO - Getting batch 38 status...
[2020-12-22 14:11:06,983] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,983] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:11:06,987] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,987] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/38
[2020-12-22 14:11:06,994] {logging_mixin.py:112} INFO - [2020-12-22 14:11:06,994] {batch.py:90} INFO - Batch 38 has not finished yet (state is 'starting')
[2020-12-22 14:11:27,013] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,013] {batch.py:80} INFO - Getting batch 38 status...
[2020-12-22 14:11:27,020] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,020] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:11:27,024] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,024] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/38
[2020-12-22 14:11:27,031] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,031] {batch.py:299} INFO - --------------------------------------------------Full log for batch 38--------------------------------------------------
[2020-12-22 14:11:27,039] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,039] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:11:27,043] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,043] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/38/log?from=0&size=100
[2020-12-22 14:11:27,050] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,050] {batch.py:309} INFO - stdout: 
[2020-12-22 14:11:27,053] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,053] {batch.py:309} INFO - 2020-12-22 14:11:08,736 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:11:27,055] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,055] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:11:27,055] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,055] {batch.py:309} INFO -   File "/tmp/spark-b71ae638-0706-4ec0-9e68-4bc51cab0b45/main.py", line 86, in <module>
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO -     main()
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO -   File "/tmp/spark-b71ae638-0706-4ec0-9e68-4bc51cab0b45/main.py", line 56, in main
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO -     spark_files_dir = SparkFiles.getRootDirectory()
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO -   File "/usr/spark/python/lib/pyspark.zip/pyspark/files.py", line 59, in getRootDirectory
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO - AttributeError: 'NoneType' object has no attribute '_jvm'
[2020-12-22 14:11:27,056] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO - 2020-12-22 14:11:11,869 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:11:27,057] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,056] {batch.py:309} INFO - 2020-12-22 14:11:11,870 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-b71ae638-0706-4ec0-9e68-4bc51cab0b45
[2020-12-22 14:11:27,057] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,057] {batch.py:309} INFO - 2020-12-22 14:11:11,876 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:11:27,057] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,057] {batch.py:309} INFO - 2020-12-22 14:11:11,877 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:11:27,057] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,057] {batch.py:309} INFO - 2020-12-22 14:11:11,877 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:11:27,057] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,057] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:11:27,058] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,057] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:11:27,058] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,058] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:11:27,058] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,058] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 38--------------------------------------------------
[2020-12-22 14:11:27,058] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,058] {batch.py:335} INFO - Closing batch with id = 38
[2020-12-22 14:11:27,069] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,069] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:11:27,072] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,072] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/38
[2020-12-22 14:11:27,079] {logging_mixin.py:112} INFO - [2020-12-22 14:11:27,079] {batch.py:340} INFO - Batch 38 has been closed
[2020-12-22 14:11:27,087] {taskinstance.py:1128} ERROR - Batch 38 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 38 failed with state 'dead'
[2020-12-22 14:11:27,090] {taskinstance.py:1170} INFO - All retries failed; marking task as FAILED.dag_id=loan_prediction_train_model, task_id=preprocess_data, execution_date=20201222T124323, start_date=20201222T141106, end_date=20201222T141127
[2020-12-22 14:11:31,850] {logging_mixin.py:112} INFO - [2020-12-22 14:11:31,850] {local_task_job.py:103} INFO - Task exited with return code 1
